### ISSUE TITLE
Setting up ezsql

### ISSUE THREAD

**Issue by usabilitest** [ezSQL/ezsql#151]
First I would like to mention that I am very familiar with ezsql. I've used the older version for years but while trying to get it started on a new project I ran into some issues that require me to switch to the latest version. Unfortunately the latest version relies on composer, which I purposefully avoided till now. Regardless, I went ahead and installed composer and ran ezsql install with it. This is as far as I was able to get. New version lacks any examples on how to proceed with the setup.

I was able to locate where ezsql was placed by composer, but it seem it pulled v.3.1 and not the latest. 

I've included require '../../../vendor/autoload.php'; but am still unclear where I would store my db credentials.

What do I do next? Can you please add a working example? 

Thanks.

**Comment by techno-express** [comment]
Take an look tests folder for the SQL driver you are using and some examples, and look over readme file.

Anyway, typing.
`composer require ezsql/ezsql` should have pulled version 4.

Then in your PHP file have just.
`require 'vendor/autoload.php';`

Setup database credentials like this.
`$db = Database::initialize('mysqli', [$user, $password, $database, $port]);`
or `$db = mysqlInstance([$user, $password, $database, $host, $port]);`
or `$db = pgsqlInstance([$user, $password, $database, $host, $port]);`

There after to get the created database instance at anytime call `getInstance();`

**Comment by techno-express** [comment]
If you want to use v3.1 still, then look at [phpunit tests for version 3](https://github.com/ezSQL/ezsql/tree/v3/tests). 
And composer is not required, just do `require_once "ez_sql_loader.php";` in your main file.

**Comment by usabilitest** [comment]
I tried again running composer. Here's the output I get:
```

> $ composer require ezsql/ezsql
> Using version ^3.1 for ezsql/ezsql
> ./composer.json has been updated
> Loading composer repositories with package information
> Updating dependencies (including require-dev)
> Nothing to install or update
> Generating autoload files

```

Perhaps it's in setting of the project?

**Comment by techno-express** [comment]
What version of PHP you using?

Also try deleting your current folder that it's in or use an empty folder
or delete your json and lock file. Just tried and it pulls version 4.

**Comment by usabilitest** [comment]
My PHP version is: 7.0.33

I deleted ezsql directory, deleted both .lock and .json files s well as .composer/cache/files/ezsql

Ran it again:

`$ composer require ezsql/ezsql
Using version ^3.1 for ezsql/ezsql
./composer.json has been created
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing ezsql/ezsql (3.1.2): Loading from cache
Writing lock file
Generating autoload files
$ composer require ezsql/ezsql
Using version ^3.1 for ezsql/ezsql
./composer.json has been created
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing ezsql/ezsql (3.1.2): Downloading (100%)
Writing lock file
Generating autoload files`

**Comment by techno-express** [comment]
I guess you need to pull it directly.
composer require ezsql/ezsql=^4.0.0

**Comment by usabilitest** [comment]
OK, this is weird... I tried running 4.0.0 but got the following error: 

> composer require ezsql/ezsql=^4.0.0

I wonder if I have multiple versions of PHP on the same server... Perhaps I need to reinstall composer.

**Comment by techno-express** [comment]
Yes, something is not correct.

**Comment by usabilitest** [comment]
I've attempted to reinstall composer to my public_html using instructions that I found here:
https://www.namecheap.com/support/knowledgebase/article.aspx/9977/29/how-to-install-composer-on-shared-servers
for this particular project but wasn't able to for some reason. It is always installed way higher and it appears my server does have PHP 5.6.40 and only once you get to this domain's public_html it's v7.

At this point I'm not ready to make a global PHP upgrade, so I'm was able to install ezsql v3.1.2
 I've added require_once '../../../vendor/ezsql/ezsql/ez_sql_loader.php'; 
However, when I try to initialize $db it breaks, because I believe I'm using syntax for v4

$db = Database::initialize('mysqli', [$user, $password, $database, $port]);

How do I do it in v3?

Thanks

**Comment by techno-express** [comment]
The same as before, in v2.1.7.

**Comment by techno-express** [comment]
Also if using composer, it will be require 'vendor/autoload.php', read v3
readme

**Comment by usabilitest** [comment]
It turns out that I can change PHP version through cPanel, however it only affects public_html and up. I need to figure out hot to install composer in public_html that will enable me to transition other projects, one at a time, to ezsql to version 4 that are currently using v.2.03. Simply switching PHP to version 7 will now break those services.

Unfortunately, it seems no matter what I do .composer is installed above publick_html...

**Comment by techno-express** [comment]
copy and extract zip file. but that defeats the purpose, and back to old database setup routine.

**Comment by usabilitest** [comment]
OK, I agree that it defeats the purpose. But it seems I've got catch 22 situation here. I cannot change server PHP version without breaking live site because the old ezsql won't work with PHP 7, and I can't install ezsql 4 because my server settings are too old.

I think I need to attempt a multi-step transition. If ezsql v3 works with both PHP 5.6 and 7 I need to install it, transition live site to ezsql v3 and then upgrade PHP to v7. This will allow me to install ezsql 4 and then make a final transition of the live site.

Question, does ezsql work on both versions of PHP 5.6 and 7?

If yes, I'm going to proceed. In a meantime I've re-installed composer and ezsql. I've verified that both files exist relative to my test file (/public_html/test.php), but I'm still not able to initialize the db. I've tried the  following approaches:

```
require_once '../vendor/ezsql/ezsql/ez_sql_loader.php';
$setting = new Config('mysqli', ['my_usr', 'my_pass', 'my_db', 'localhost']);
$db = new ez_mysqli($settings);
```
then I tried:
```

require_once '../../../vendor/autoload.php';
$db = Database::initialize('mysqli', ['my_usr', 'my_pass', 'my_db', 'localhost']);
```
unfortunately in both cases I got the following error:

> received error: Fatal error: Default value for parameters with a class type hint can only be NULL in /home/mydomain/vendor/ezsql/ezsql/shared/ezFunctions.php on line 247

What am I missing?
